dexonlineplugin/prototype.py

```diff
@@ -0,0 +1,364 @@
+    if token.pos_ in banned_pos:
+        return False
+    if token.lang_ != "ro":
+        return False
+    if not token.text.isalpha():
+        return False
+    if token.ent_type_ in banned_ent_types:
+        return False
+    return True
+
+def get_wanted_form(token, pos_finder, person, number):
+    """
+       This function will return the morph form wanted by pos_finder, person and number
+    """
+    all_morph = get_all_forms(token)
+    for wanted_form in all_morph:
+        if pos_finder in wanted_form['pos'] and person in wanted_form['pos'] and number in wanted_form['pos']:
+            return wanted_form['form']
+
+def verify_word_at_certain_pos(token, pos_verifier):
+    """
+    verifiy if a token is contains a specified string in its part of speech
+    for example this function will return true if a verb has this description from dexonline
+    as its pos "Verb, Indicativ, perfect simplu, persoana I, singular" and pos_verifier parameter
+    is "perfect simplu" or "persoana I", etc  
+    """
+    all_morph = get_all_forms(token)
+    for wanted_form in all_morph:
+        if token.text == wanted_form['form']:
+            for pos in pos_verifier:
+                if pos in wanted_form['pos']:
+                    return True
+
+def get_right_person_and_number(token):
+    """
+        This function will get the person and number data from token.morph and will convert these into
+        dexonline database format information in order to select right form of verb.
+    """
+    # extract correct person and number for a phrase
+    person = token.morph.get("Person") if token.morph.get("Person") else ['3']
+    number = token.morph.get("Number") if token.morph.get("Number") else ['Sing']
+    # extra step to verify if there is a composed subject (like 'eu cu tine mergem')
+    ok_switch_number = 0
+    if not token.pos_ == "VERB" and not token.pos_ == "AUX":
+        if len(list(token.children)):
+            for t in token.children:
+                if t.text not in ["m", "te", "s"]:
+                    ok_switch_number = 1 
+                    break
+    if ok_switch_number:
+            number = ["Plur"]
+
+    #formatting number and person to be recognized dexonline json
+    actual_number = "plural" if number == ["Plur"] else "singular"
+
+    if person == ['1']:
+        actual_person = "I"
+    elif person == ['2']:
+        actual_person = "II"
+    elif person == ['3']:
+        actual_person = "III"
+
+    return actual_number, actual_person
+
+def forme_reflexive_verifier(token):
+    """
+        This function will map short reflexive forms into long ones using data from reflexive_deps from util_data.py
+    """
+    word_added = token.text
+    if token.dep_ in reflexive_deps:
+        if token.morph.get("Case")[0] in ["Dat", "Acc"] and token.morph.get("Variant")[0] == "Short":
+            word_added = reflexive_short_to_long_form[token.text]
+
+    return word_added
+
+def oltenizare_worker(doc):
```

Comment:
momentan e ok sa fie tot codul in acelasi fisier, dar pe viitor trebuie sa ai in vedere sa separi procesarea de baza  (incarcarea dictionarelor etc) de aplicatiile ei (oltenizare, simplificare etc)